Code/UnitTests/FoundationTest/Containers/ArrayMapTest.cpp

```diff
@@ -7,6 +7,90 @@
 
 EZ_CREATE_SIMPLE_TEST(Containers, ArrayMap)
 {
+  EZ_TEST_BLOCK(ezTestBlock::Enabled, "Iterator")
+  {
+    ezArrayMap<ezUInt32, ezUInt32> m;
+    for (ezUInt32 i = 0; i < 1000; ++i)
+      m[i] = i + 1;
+
+    //{ // No finding by key (or value) using std::find without a custom comparator functor (same as in MapTest)
+    //  auto itfound = std::find(begin(m), end(m), 500);
+    //  EZ_TEST_INT(itfound->key, 499);
+    //}
+
+    //{ // No finding by key (or value) using std::find without a custom comparator functor (same as in MapTest)
+    //  auto itfound = std::find(rbegin(m), rend(m), 500);
+    //  EZ_TEST_INT(itfound->key, 499);
+    //}
```

Comment:
Is this supposed to get activated?

Reply:
I don't know! I saw the same code commented out in the MapTest.cpp (Lines 16 & 20), so I added them here as well. I thought they were left commented out there so that e.g. if there's a push to add the comparators it's a little more obvious where they should be tested or something. In any case, I just wanted to mirror the code there.

If you'd rather I just leave it out I can of course do that, too.

Reply:
Yeah, remove it. Or try if it would compile now. The comment in the map test is probably very, very old. Maybe it would even compile now? It might work since we added begin/end for ezMap.

Reply:
No it doesn't (it can't, not even in the std:::maps, and I tried): std::find runs an equality comparison between the value you pass and each element in the iterable, but since the iterable elements are pairs in a map, you can only either a) pass a kv-Pair (it'll find that of course), or b) pass in a key or value AND a functor that does the comparison correctly.

It would of course be possible to add the functor—and/or a Find method that searches for values instead of keys... ¯\_(ツ)_/¯

Reply:
Then just get rid of it.